crates/cast/bin/cmd/subscribe.rs

```diff
@@ -0,0 +1,90 @@
+use cast::{filters::build_filter, Cast};
+use clap::Parser;
+use ethers::{
+    providers::Middleware,
+    types::{BlockId, NameOrAddress},
+};
+use foundry_cli::{opts::EthereumOpts, utils};
+use foundry_config::Config;
+use std::{io, str::FromStr};
+
+/// CLI arguments for `cast subscribe`.
+#[derive(Debug, Parser)]
+pub struct SubscribeArgs {
+    /// The block height to start query at.
+    ///
+    /// Can also be the tags earliest, finalized, safe, latest, or pending.
```

Comment:
It it worthwhile to abstract this to a new FilterOpts? Since properties are the same for `cast logs` and `cast subscribe`.